src/code_7bb60_len_41b0.c

```diff
@@ -121,71 +140,117 @@ void set_action_state(s32 actionState) {
     if (actionState == ActionState_HIT_HAZARD || actionState == ActionState_HIT_LAVA) {
         u8 partner;
 
-        if (player_status->unk_BF == 3) {
+        if (playerStatus->unk_BF == 3) {
             actionState = ActionState_HIT_HAZARD;
         }
 
         // Whilst Lakilester, Bow, or Parakarry's ability is active, hazards have no effect.
-        partner = player_data->currentPartner;
-        if ((partner - 7) < 2u || (s8)partner == PartnerId_PARAKARRY) {
-            if (D_8010EBB0) {
-                player_status->animFlags |= 0x4;
-                player_status->flags |= 0x800;
+        partner = playerData->currentPartner;
+        if (((u8)(partner - 7) < 2) || (playerData->currentPartner == PartnerId_PARAKARRY)) {
+            if (D_8010EBB0[0]) {
+                playerStatus->animFlags |= 0x4;
+                playerStatus->flags |= 0x800;
                 return;
             }
         }
     }
 
     if (actionState == ActionState_SLIDING) {
-        player_status->flags |= 0x10;
-        player_status->moveFrames = 0;
-        player_status->flags &= ~0x4000;
+        playerStatus->flags |= 0x10;
+        playerStatus->moveFrames = 0;
+        playerStatus->flags &= ~0x4000;
     }
 
-    player_status->prevActionState = player_status->actionState;
+    playerStatus->prevActionState = playerStatus->actionState;
     if (actionState == ActionState_USE_TWEESTER) {
-        player_status->prevActionState = ActionState_IDLE;
+        playerStatus->prevActionState = ActionState_IDLE;
     }
 
     if (actionState == ActionState_ENEMY_FIRST_STRIKE) {
-        player_status->animFlags |= 4;
+        playerStatus->animFlags |= 4;
     }
-    player_status->actionState = actionState;
-    player_status->flags |= 0x80000000;
+    playerStatus->actionState = actionState;
+    playerStatus->flags |= 0x80000000;
 
-    if (player_status->actionState == ActionState_SPIN) {
+    if (playerStatus->actionState == ActionState_SPIN) {
         return;
     }
 
-    player_status->flags &= 0xFFFDFFFF;
-    player_status->animFlags &= 0xFFFEFFFF;
+    playerStatus->flags &= ~0x20000;
+    playerStatus->animFlags &= ~0x10000;
 
-    if (unknown_struct[0xC]) {
-        stop_sound(unknown_struct[0xC]);
+    if (unknownStruct[0xC]) {
+        stop_sound(unknownStruct[0xC]);
     }
 
-    if (player_status->unk_D8) {
-        player_status->unk_D8[3][9] = 0xA;
-        player_status->unk_D8 = NULL;
+    if (playerStatus->unk_D8) {
+        playerStatus->unk_D8[3][9] = 0xA;
+        playerStatus->unk_D8 = NULL;
     }
 }
 
 INCLUDE_ASM(s32, "code_7bb60_len_41b0", update_locomotion_state);
-
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", start_falling);
-
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", start_bounce_a);
-
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", start_bounce_b);
+/*void update_locomotion_state(void) {
```

Comment:
What's up with this? At least a comment should be left, but if its functionally equivalent, ifdef NON_MATCHING, preferably.

Reply:
Oops, that was supposed to be removed from the PR.